### PR TITLE
PHP 8 support

### PR DESCRIPTION
**Pull request by adlenton**
Hey,

thank you for picking up maintenance for Assetic!
As we are using the framework and are preparing for the PHP 8 upgrade I took the liberty to apply some changes needed for this move.

Changes:
- updated php & php unit version
- compatibility changes for php unit upgrade
- minor cleanups I saw on my way there

### CODE REVIEW

**Comment by jaxwilko** [comment]
Had a quick check and all looks good, I'll do a proper review and try and respond before the weekend. 

Thanks for the PR 😁

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
+      <directory suffix="Test.php">./tests/Assetic/Test/</directory>
+    </testsuite>
+  </testsuites>
+  <php>
+    <server name="NODE_PATH" value="./node_modules/"/>
+    <!-- <server name="CLOSURE_JAR" value="/path/to/google-closure/compiler.jar" /> -->
+    <!-- <server name="CSSEMBED_JAR" value="/path/to/cssembed.jar" /> -->
+    <!-- <server name="GSS_JAR" value="/path/to/closure-stylesheets.jar" /> -->
+    <!-- <server name="SPROCKETS_LIB" value="/path/to/sprockets/lib" /> -->
+    <!-- <server name="YUI_COMPRESSOR_JAR" value="/path/to/yuicompressor-2.4.2.jar" /> -->
```
Comment (LukeTowers) [inline phpunit.xml.dist]:
```suggestion
```

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,25 +1,16 @@
...
+    </include>
+  </coverage>
+  <testsuites>
+    <testsuite name="Assetic Test Suite">
+      <directory suffix="Test.php">./tests/Assetic/Test/</directory>
+    </testsuite>
+  </testsuites>
+  <php>
+    <server name="NODE_PATH" value="./node_modules/"/>
+  </php>
```
Comment (LukeTowers) [inline phpunit.xml.dist]: Please keep the original four spaces indentation

Comment (adlenton) [reply phpunit.xml.dist]: Fixed, thanks for the hint!

**Comment by adlenton** [comment]
I reverted some changes of assertions that are deprecated in the new phpunit version so that we can still support php 7.2

**Comment by brainexe** [comment]
@jaxwilko any update here? :) I'm also waiting for this change as it somehow prevents us from starting the full php8 update

**Comment by LukeTowers** [comment]
LGTM!

**Pull request merged**